src/collectors/processcheck/processcheck.py

```diff
@@ -0,0 +1,183 @@
+        """
+        self.processes = {}
+        self.processes_info = {}
+        for pg_name, cfg in self.config['process'].items():
+            pg_cfg = {}
+            for key in ('exe', 'name', 'cmdline'):
+                pg_cfg[key] = cfg.get(key, [])
+                if not isinstance(pg_cfg[key], list):
+                    pg_cfg[key] = [pg_cfg[key]]
+                pg_cfg[key] = [re.compile(e) for e in pg_cfg[key]]
+            self.processes[pg_name] = pg_cfg
+            self.processes_info[pg_name] = {}
+
+    def get_default_config_help(self):
+        config_help = super(ProcessCheckCollector,
+                            self).get_default_config_help()
+        config_help.update({
+            'process': ("A subcategory of settings inside of which each "
+                        "collected process has it's configuration"),
+        })
+        return config_help
+
+    def get_default_config(self):
+        """
+        Default settings are:
+            path: 'process'
+        """
+        config = super(ProcessCheckCollector, self).get_default_config()
+        config.update({
+            'path': 'process',
+            'process': {},
+        })
+        return config
+
+    def save_process_info(self, pg_name, process_info):
+        for key, value in process_info.iteritems():
+            if key in self.processes_info[pg_name]:
+                self.processes_info[pg_name][key] += value
+            else:
+                self.processes_info[pg_name][key] = value
+
+    def collect_process_info(self, process):
+        try:
+            pid = get_value(process, 'pid')
+            name = get_value(process, 'name')
+            cmdline = get_value(process, 'cmdline')
+            try:
+                exe = get_value(process, 'exe')
+            except psutil.AccessDenied:
+                exe = ""
+            for pg_name, cfg in self.processes.items():
+                if match_process(pid, name, cmdline, exe, cfg):
+                    pi = {}
+                    status = get_value(process, 'status')
+                    self.log.debug("process %s has its status as %s", pg_name, status)
+                    pi.update({'up': 1 if status not in [psutil.STATUS_DEAD, psutil.STATUS_STOPPED] else 0})
+                    self.log.debug("process %s has its process info as %s", pg_name, pi)
+                    self.save_process_info(pg_name, pi)
+        except psutil.NoSuchProcess, e:
+            self.log.info("Process exited while trying to get info: %s", e)
+
+    def collect(self):
+        """
+        Check each process defined under the
+        `process` subsection of the config file
+        """
+        if not psutil:
+            self.log.error('Unable to import psutil, no process check performed')
+            return None
+
+        for process in psutil.process_iter():
+            self.collect_process_info(process)
+
+        # check results
+        for pg_name, counters in self.processes_info.iteritems():
```

Comment:
Why store `processes_info` in a global instead of returning it four lines above and iterating through that result? Modification and iteration of a global seems unnecessarily complicated.

Reply:
I reused a lot of code from ProcessResourcesCollector, that's the answer to a lot of your questions.

Reply:
I think code requirements are fewer to just check if a process exists. You should be able to cut out a couple of methods and simplify the whole collector.

Reply:
sounds good